### PR TITLE
Factored out QuestionFontSizeProvider

### PR DESCRIPTION
**Pull request by grzesiek2010**
#### What has been done to verify that this works as intended?
I reviewed implemented changes and tested changing font size manually.

#### Why is this the best possible solution? Were any other approaches considered?
@seadowg 
In most cases we use object we pass/inject. In this case I just created an utility class with a static method because I don't think it's something we need test (mock) so this approach is easier and we will avoid passing many object via constructors.
...but let me know if your opinion differs.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It shouldn't affect/change anything it's refactoring that should just make the code a bit cleaner. Quick testing changing font size would be enough.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/utilities/QuestionFontSizeProvider.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package org.odk.collect.android.utilities;
+
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+
+import static org.odk.collect.android.preferences.GeneralKeys.KEY_FONT_SIZE;
+
+public class QuestionFontSizeProvider {
```
Comment (seadowg) [inline collect_app/src/main/java/org/odk/collect/android/utilities/QuestionFontSizeProvider.java]: This should probably just be `QuestionFontSizeUtils` now

**Review by seadowg — APPROVED** [review]
I was in two minds about this being static or not. I think if we were testing that a widget is using it (which we really should be) we'd probably want to write something like this:

```java
assertThat(widget.getSomeTextView().getTextSize(), equalTo(QuestionFontSizeUtils.getQuestionFontSize());
```

This seems nicer than faking the font size then asserting on the fake or (worse still) verifying we call `getQuestionFontSize` so the static feels right here.

File: collect_app/src/main/java/org/odk/collect/android/utilities/QuestionFontSizeProvider.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/utilities/QuestionFontSizeProvider.java]: Right, fixed.

**Review by seadowg — APPROVED** [review]

**Comment by mmarciniak90** [comment]
**Tested with success**

> It shouldn't affect/change anything it's refactoring that should just make the code a bit cleaner. Quick testing changing font size would be enough.

I can confirm that I didn't notice any change.

Verified on Android: 6.0, 7.0, 8.1, 9.0
Verified on an emulator with Android 5.1

Verified cases:
* change the font size before start filling a form
* change the font size while filling a form
* scan QR code with extra small font size (without emulator)
* range picker reacts on size change
* `All widgets` form

@getodk-bot unlabel "needs testing"
@getodk-bot label "behavior verified"

**Pull request merged**